src/main/java/frc/robot/subsystems/Swerve/SwerveSubsystem.java

```diff
@@ -1,47 +1,51 @@
 package frc.robot.subsystems.Swerve;
 
 import static frc.robot.Constants.Constants.*;
-
-import java.util.Optional;
-
-import edu.wpi.first.math.VecBuilder;
-import edu.wpi.first.math.estimator.SwerveDrivePoseEstimator;
-
+import static frc.robot.RobotContainer.POSE_ESTIMATOR;
+import org.littletonrobotics.junction.Logger;
 import edu.wpi.first.math.geometry.Translation2d;
 import edu.wpi.first.math.kinematics.ChassisSpeeds;
 import edu.wpi.first.math.kinematics.SwerveDriveKinematics;
 import edu.wpi.first.math.kinematics.SwerveModulePosition;
 import edu.wpi.first.math.kinematics.SwerveModuleState;
-import edu.wpi.first.networktables.GenericEntry;
-import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj.shuffleboard.BuiltInWidgets;
 import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
-import edu.wpi.first.wpilibj.smartdashboard.Field2d;
+import edu.wpi.first.wpilibj.shuffleboard.SimpleWidget;
 import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
 import edu.wpi.first.wpilibj2.command.Command;
 import edu.wpi.first.wpilibj2.command.InstantCommand;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
-import frc.robot.LimelightHelpers;
 import frc.robot.Constants.SwerveModuleConfiguration;
 
-import frc.robot.subsystems.IMU;
-import edu.wpi.first.math.geometry.Pose2d;
-import edu.wpi.first.math.geometry.Rotation2d;
-
-
-
+import java.util.Map;
 
 public class SwerveSubsystem extends SubsystemBase {
-	
-	
-	private GenericEntry currentPos = Shuffleboard.getTab("swerve").add("curr_pos",new double[3]).getEntry();
-	
-	
-	private IMU imu = new IMU();
-
-	public SwerveDrivePoseEstimator pose_est;
-
-	Field2d field = new Field2d();
-	
+	public void velocityGraphUpdate(double xVelocity, double yVelocity){
```

Comment:
All SmartDashboard code related with controls should be in the InputManger instead of Swerve. These values are not use in the SwerveSubsystem class

Reply:
@AreaProgram I know you wrote this and @GoofyCoderGuy I know you made the input manager, can you work together to bring this into the inputmanager?

Reply:
I don't think it will make much sense to put this in the input manager since it doesn't deal with the controller/joystick at all. Rather it deals with the velocities calculated by the default drive class. So wouldn't it make more sense to put that function in there?

Reply:
I was thinking the same thing as Matthew, as the input manager deals with the controller and the joystick values, not the velocities. I can put it into a different file, but that seemed like the best one when I made it because there was no dedicated shuffleboard file and other files have their shuffleboard data in their respective file.

Reply:
using the values in the swerve drive code is an option, and actually I think is more clear that these limit the swerve max speeds. I leave which version to do up to you. But the current code mission state between classes is not ideal.

Reply:
I think it would be better to have this function in another file. It might be better altogether to create a new shuffleboard dedicated file that extends SubvsystemBase or as an extension to Input Manager to handle all of the shuffleboard inputs, as I know that there are a lot of limelight values that use Network tables and Shuffleboard. We can maybe even have advantageScope code in there for testing purposes. If you do not think it is necessary now, I'd be fine with putting it in input manager for now.

Reply:
SGTM